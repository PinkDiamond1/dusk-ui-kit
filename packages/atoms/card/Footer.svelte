<div class="{$$props.class || ''} duk-card__footer">
  <slot />
</div>
