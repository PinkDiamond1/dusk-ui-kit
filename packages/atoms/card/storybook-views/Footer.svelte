<script>
  import Card, { Content, Footer } from "../";
  export let variant;
  export let elevation;
  export let content;
  export let size;
  export let tone;
</script>

<Card {variant} {tone} {elevation} {size} style="width: 305px">
  <Content>{content}</Content>
  <Footer>
    <p>This is some footer text.</p>
    <p>This is some footer text.</p>
  </Footer>
</Card>
