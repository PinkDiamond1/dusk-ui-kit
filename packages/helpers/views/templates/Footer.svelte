<script>
  import { current_component } from "svelte/internal";
  import { forwardEventsBuilder, exclude, useActions } from "@dusk/helpers";
  const forwardEvents = forwardEventsBuilder(current_component);
  export let use = [];
  let className = "";
  export { className as class };
</script>

<footer
  use:useActions="{use}"
  use:forwardEvents
  class="duk-template__footer {className}
  "
  {...exclude($$props, ['use', 'class'])}>
  <slot />
</footer>
