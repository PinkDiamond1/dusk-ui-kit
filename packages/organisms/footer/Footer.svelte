<script>
  import { current_component } from "svelte/internal";
  import {
    forwardEventsBuilder,
    exclude,
    useActions,
    variants,
    types,
  } from "@dusk-network/helpers";
  import Minimal from "./Minimal.svelte";
  import Maximal from "./Maximal.svelte";
  import "./styles.css";

  let className = "";
  export { className as class };
  export let variant = variants.ORGANISM.FOOTER.BRAND;
  export let type = types.ORGANISM.FOOTER.MINIMAL;
  export let use = [];
  export let social;
  export let categories;

  let today = new Date();
  let year = today.getFullYear();
  const copyright = `Â© 2018 - ${year} Dusk Network B.V. All Rights Reserved.`;

  const forwardEvents = forwardEventsBuilder(current_component);

  function getClassNames(variant) {
    let classNames = "";
    switch (variant) {
      case variants.ORGANISM.FOOTER.BRAND:
        classNames += " duk-footer--brand";
        break;
      case variants.ORGANISM.FOOTER.CTA:
        classNames += " duk-footer--cta";
        break;
      case variants.ORGANISM.FOOTER.INFO:
        classNames += " duk-footer--info";
        break;
      case variants.ORGANISM.FOOTER.SUCCESS:
        classNames += " duk-footer--success";
        break;
      case variants.ORGANISM.FOOTER.WARNING:
        classNames += " duk-footer--warning";
        break;
      case variants.ORGANISM.FOOTER.DANGER:
        classNames += " duk-footer--danger";
        break;
      case variants.ORGANISM.FOOTER.LIGHT:
        classNames += " duk-footer--light";
        break;
      case variants.ORGANISM.FOOTER.DARK:
        classNames += " duk-footer--dark";
        break;
      case variants.ORGANISM.FOOTER.BLACK:
        classNames += " duk-footer--black";
        break;
      case variants.ORGANISM.FOOTER.WHITE:
        classNames += " duk-footer--white";
        break;
      default:
        classNames += "";
    }

    return classNames;
  }
</script>

<footer
  use:useActions="{use}"
  use:forwardEvents
  class="duk-footer {className} {getClassNames(variant)}"
  {...exclude($$props, ['use', 'class', 'type', 'variant', 'social', 'categories'])}>
  {#if type === types.ORGANISM.FOOTER.MAXIMAL}
    <Maximal categories="{categories}" />
  {/if}
  <Minimal copyright="{copyright}" social="{social}" />
</footer>
