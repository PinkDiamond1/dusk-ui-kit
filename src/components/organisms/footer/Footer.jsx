/* @flow */
import * as React from 'react'

import './Footer.scss'

export const FooterVariants = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  DANGER: 'danger',
}

type Props = {
  className: string,
  children: React.Node,
  variant?: string,
}

const Footer = (props: Props): React.Element<*> => {
  const { className, children, variant } = props
  let updatedClassname = `__duk-footer ${className} `
  switch (variant) {
    case 'primary':
      updatedClassname += `bg-primary text-white`
      break
    case 'secondary':
      updatedClassname += `bg-secondary text-white`
      break
    case 'info':
      updatedClassname += `bg-info text-white`
      break
    case 'success':
      updatedClassname += `bg-success text-white`
      break
    case 'warning':
      updatedClassname += `bg-warning text-white`
      break
    case 'danger':
      updatedClassname += `bg-danger text-white`
      break
    default:
      updatedClassname = `bg-primary text-white`
  }
  return <footer className={updatedClassname}>{children}</footer>
}

Footer.defaultProps = {
  className: '',
  variant: FooterVariants.PRIMARY,
}

export default Footer
