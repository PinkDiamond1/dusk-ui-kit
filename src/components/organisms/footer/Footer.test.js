import React from 'react'
import renderer from 'react-test-renderer'

import Footer from './Footer'

it('test to see if the Footer renders correctly', () => {
  const tree = renderer.create(<Footer>Smoke Test</Footer>).toJSON()
  expect(tree).toMatchSnapshot()
})
