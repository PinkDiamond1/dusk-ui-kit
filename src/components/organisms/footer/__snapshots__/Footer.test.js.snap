// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Footer renders correctly 1`] = `
<footer
  className="__duk-footer  bg-primary text-white"
>
  Smoke Test
</footer>
`;
